src/Block/Ssr/RecordList.php

```diff
@@ -82,7 +82,7 @@ protected function _afterToHtml($html)
     protected function searchResult(RequestInterface $request, array $searchParams): array
     {
         $paramsString = implode('&', array_filter([
-                parse_url($request->getUriString(), PHP_URL_QUERY),
+                parse_url($request->getUriString() ?? '', PHP_URL_QUERY),
```

Comment:
can you explain this change?

Reply:
Yes, I have failed integration tests on my local instace. 
![image](https://user-images.githubusercontent.com/11246832/155494393-11c8a92f-a052-4876-a72f-534c2b8e66aa.png)

Reply:
i don't think changing implementation in that way to fit test is a good approach.  You should rather mock the request to return something after calling `getUriString` on it

Reply:
But didn't this test shows that system can fails because of this error? But, according to docs, method `getUriString` can return only string and I'm wandering how it return null in this case. Maybe I don't quite understand this case

Reply:
it might fail but only in artificially created environment i think. 
Let' leave this file for now. From what i see this test should be changed in order to guarantee that request object will be correctly created by the test framework

Reply:
OK, I'll rollback this change because according to docs method can return only string value so this changes looks like useless